File: crates/enforcer-memory/tests/fixtures/memory/lang_perl/widget.pl
use strict;
use warnings;
use POSIX qw(floor);

package Widget;
require Exporter;

sub new {
    my $class = shift;
    my $self = { name => shift };
    bless $self, $class;
    return $self;
}

sub draw {
    my $self = shift;
    if ($self->{name}) {
        print("drawing: " . $self->{name});
    } else {
        print("drawing: unnamed");
    }
    return helper($self);
}

sub helper {
    my $widget = shift;
    return $widget->{name};
}

my $w = Widget->new("box");
$w->draw();

File: crates/enforcer-memory/tests/fixtures/memory/lang_clojure/widget.clj
(ns app.widget
  (:require [clojure.string :as str]))

(defrecord Widget [name])

(defn helper [x y]
  (+ x y))

(defn draw [widget]
  (if (> (helper 1 2) 0)
    (println (str/join " " ["drawing" (:name widget)]))
    (println "drawing: nothing"))
  (helper 3 4))

(defn main []
  (draw (Widget. "box")))
